tests/src/unit-bjdata.cpp

```diff
@@ -2488,6 +2488,27 @@ TEST_CASE("BJData")
                 CHECK_THROWS_WITH(_ = json::from_bjdata(v), "[json.exception.parse_error.112] parse error at byte 4: syntax error while parsing BJData size: expected '#' after type information; last byte: 0x02");
             }
 
+            SECTION("optimized array: negative size")
+            {
+                std::vector<uint8_t> v1 = {'[', '#', 'i', 0xF1};
+                std::vector<uint8_t> v2 = {'[', '$', 'I', '#', 'i', 0xF2};
+                std::vector<uint8_t> v3 = {'[', '$', 'I', '#', '[', 'i', 0xF4, 'i', 0x02, ']'};
+                std::vector<uint8_t> v4 = {'[', '$', 0xF6, '#', 'i', 0xF7};
+                std::vector<uint8_t> v5 = {'[', '$', 'I', '#', '[', 'i', 0xF5, 'i', 0xF1, ']'};
+                std::vector<uint8_t> v6 = {'[', '#', '[', 'i', 0xF3, 'i', 0x02, ']'};
+
+                json _;
+                CHECK_THROWS_AS(_ = json::from_bjdata(v1), json::out_of_range&);
```

Comment:
Please us `CHECK_THROWS_WITH_AS` macro (see https://github.com/doctest/doctest/blob/master/doc/markdown/assertions.md#exceptions) to check both the exception type and the diagnostic output.

Reply:
do you prefer me to restore this (and merge `WITH` and `AS`)? the out-of-range bound seems to be platform dependent

https://github.com/NeuroJSON/json/commit/ffcf148667ddd6213d3d31958f10f7f2680374f8#diff-9c29b8711262aa7d3b188dc44aac4ba9f898a8702af3cbde1a4946191b064294L2506-L2519

msvc [did not like the condition](https://github.com/nlohmann/json/runs/6344160097?check_suite_focus=true#step:5:36), but I think I can fix that.

Reply:
I think you should restore those checks and fix the warning (by suppressing it with doctest macros if need be).